ddtrace/contrib/internal/mcp/patch.py

```diff
@@ -230,6 +233,48 @@ async def traced_client_session_aexit(mcp, pin: Pin, func, instance, args: tuple
             span.finish()
 
 
+@with_traced_module
+def traced_request_responder_enter(mcp, pin: Pin, func, instance, args: tuple, kwargs: dict):
+    integration: MCPIntegration = mcp._datadog_integration
+    request = instance.request.root
+
+    # While this patch can trace all requests, we only trace this type right now
+    if not isinstance(request, InitializeRequest):
+        return func(*args, **kwargs)
+
+    span = integration.trace(
+        pin, REQUEST_RESPONDER_ENTER_OPERATION_NAME, submit_to_llmobs=True, span_name="mcp.initialize"
+    )
+    instance._dd_span = span
+    integration.llmobs_set_tags(
+        span,
+        args=[instance] + list(args),
+        kwargs=kwargs,
+        response=None,
+        operation=REQUEST_RESPONDER_ENTER_OPERATION_NAME,
+    )
```

Comment:
do we need this here if we do it in `responder.respond` below?

Reply:
I could try to combine them. Right now _enter processes the request and `respond` processes the response.

Reply:
Ok, I spent some time trying to combine these, and I'm not sure how I should handle intent capture in the case that we cannot have multiple set_tags methods. If I process the intent after the handler responds, then it means the tool will have access to the tracing parameters. In the Go implementation, and the python implementation I had prepared, the intent parameter is deleted before the handler is called. Are you sure we cannot have multiple set_tags?